model/src/main/java/com/javahelp/model/user/UserPassword.java

```diff
@@ -0,0 +1,91 @@
+package com.javahelp.model.user;
+
+import java.util.Base64;
+
+/**
+ * A class for the password of a user entity
+ */
+public class UserPassword {
```

Comment:
I admit that I am not too familiar with salt sand hash and things but, do we need any common attributes between the User class and the UserPassword class in order to match a UserPassword instance with the specific user? For example, do we need an "ID" attribute in the UserPassword class so that we know this password instance is for the user of that ID?

Reply:
I was thinking that could more be handled by the API gateway. I.E. you pass in a user id and get the associated `UserPassword`. Instead of storing the associated ID in the `UserPassword`, this detail can be handled by the database, and by implementations of the API gateway